src/olympia/blocklist/tasks.py

```diff
@@ -143,16 +148,37 @@ def import_block_from_blocklist(record):
 
 
 @task
-def upload_filter_to_kinto(generation_time):
+def upload_filter_to_kinto(generation_time, is_base=True, upload_stash=False):
     server = KintoServer(
         KINTO_BUCKET, KINTO_COLLECTION_MLBF, kinto_sign_off_needed=False)
+    mlbf = MLBF(generation_time)
+    # Deal with possible stashes first
+    if upload_stash:
+        # If we have a stash, write the attachment for that
+        stash_data = {
+            'key_format': MLBF.KEY_FORMAT,
+            'stash_time': generation_time,
+            'attachment_type': BLOCKLIST_RECORD_STASH,
+        }
+        with storage.open(mlbf.stash_path, 'r') as stash_file:
+            attachment = ('stash.json', stash_file)
+            server.publish_attachment(stash_data, attachment)
+    if is_base:
+        # clear the collection for the base - we want to be the only filter
+        server.delete_all_records()
```

Comment:
If `is_base == upload_stash == True`, then you would be uploading the stash and immediately delete it. I think that you have to delete before pushing any new records/attachments.

Reply:
Ah, you're right.  In current use `upload_stash` is always `not is_base` so it wouldn't happen but I should fix that.

Reply:
https://github.com/mozilla/addons-server/pull/13986/commits/a79c82f67ed89a36986fc2acd872e95db48ffc34